pyteal/ast/maybe.py

```diff
@@ -27,8 +31,22 @@ def __init__(
             immediate_args (optional): Immediate arguments for the op. Defaults to None.
             args (optional): Stack arguments for the op. Defaults to None.
         """
+
+        def local_version_check(option: "CompileOptions"):
```

Comment:
Was the original intent to do a compile check based on the fact that `MultiValue()` was being employed for these TEAL8 opcodes? Now that `MaybeValue` is being used instead, does it still make sense to have `MultiValue` have this verstion checking, or should it all be done in `MaybeValue` ?

Reply:
> Was the original intent to do a compile check based on the fact that `MultiValue()` was being employed for these TEAL8 opcodes?

If I am understanding correctly, do you mean the purpose of introducing such compile check is because `MultiValue` was the first implementation of these `box` related `Teal8` opcodes?

If so, then yes. I think `MaybeValue`, which is inherited from `MultiValue`, does not have a compile check, seems flawed.

> Now that `MaybeValue` is being used instead, does it still make sense to have `MultiValue` have this version checking, or should it all be done in `MaybeValue` ?

I believe this it is still making sense, for `MaybeValue` is a 2-return-value special case in general `MultiValue` settings.

I might be understanding it wrongly, but if you are suggesting all the version checking in `MaybeValue` can be applying to all cases in `MultiValue`, I think that will hurt some implementation in `ecdsa`, for they use `verifyFieldVersion` rather than `verifyTealVersion`. Such flexibility seems to be still needed?